### ISSUE TITLE
Make js assets comply to javascript strict mode

### ISSUE THREAD

**Issue by mikehaertl** [yiisoft/yii2#12580]
**tl;dr** 
 * Replace `yii = (function ($) {` with `window.yii = (function ($) {` in `assets/yii.js`
 * This is **not** about adding `"use strict;"` to yii assets!

---- 
I'm currently implementing a build chain for my custom javascript modules in [ES2015](https://babeljs.io/docs/learn-es2015/). I use browserify, babelify and minifyify to transpile, concat and minify all my scripts to a single file. I also want to include some bundled yii assets like `yii.js` or `yii.activeForm.js`. 

Unfortunately `yii.js` is not strict mode compliant:

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Strict_mode

This means, you can not concatenate it with other sources, that use `"use strict;"` somewhere - which is what browserify does. The main problem is, that `yii.js` uses an undefined variable in global scope:

```
yii = (function ($) { ...
```

This leads to `Uncaught ReferenceError: yii is not defined` in the browser. This could simply be fixed by doing this instead:

```
window.yii = (function ($) { ...
```

Alternatively a `var` could be added: `var yii = ...`.  It's **not** required and **not recommended** that `yii.js` also gets a `"use strict;"` statement. This way the file can be used in both modes: strict and non-strict as it complies to both standards.

For the record: I've also opened an issue at [browserify-shim](https://github.com/thlorenz/browserify-shim/issues/220), but I think this should rather be fixed on the yii side.

**Comment by rob006** [comment]
This is already fixed by #12584
